DragaliaAPI/DragaliaAPI/Services/Game/StoryService.cs

```diff
@@ -256,6 +261,16 @@ private async Task<List<AtgenBuildEventRewardEntityList>> ReadQuestStory(int sto
             );
         }
 
+        if (storyId == 1001009)
```

Comment:
I wonder if it's possible to do all this by extending the generic reward check above? (L218)

Reply:
I initially added the rewards to QuestStoryRewardInfo.json, but the rewards were presented in the quest reward screen rather than the gift box.

I could remove the Dictionary if I do that if I go this route.
```c#
if (reward.Type == EntityTypes.FortPlant)
{
      await fortRepository.AddToStorage((FortPlants)reward.Id, reward.Quantity, true);
}
else if (storyId == 1001009)
{
      presentService.AddPresent(
            new Present(
                 PresentMessage.Chapter10Clear,
                 (EntityTypes)reward.Type,
                 reward.Id,
                 reward.Quantity
             )
        );
}
else
{
...
``` 
The problem here is that the story reward message mentions that the presents will go to the "Limited" box, but they actually go to the "No Limit" box. I haven't found a way to move these to that box correctly. 🤔